exec/java-exec/src/main/java/org/apache/drill/exec/store/ImplicitColumnUtils.java

```diff
@@ -0,0 +1,141 @@
+      return implicitColumns.get(fieldName);
+    }
+
+    public void writeImplicitColumns() {
+      ImplicitColumn column;
+      ScalarWriter writer;
+      MinorType dataType;
+      Object value;
+
+      for (Map.Entry<String, ImplicitColumn> columnEntry : implicitColumns.entrySet()) {
+        column = columnEntry.getValue();
+        writer = column.writer;
+        dataType = column.dataType;
+        value = column.value;
+
+        switch (dataType) {
+          case INT:
+            writer.setInt((Integer) value);
+            break;
+          case BIGINT:
+            writer.setLong((Long) value);
+            break;
+          case FLOAT4:
+            writer.setFloat((Float) value);
+            break;
+          case FLOAT8:
+            writer.setDouble((Double) value);
+            break;
+          case VARCHAR:
+            writer.setString((String) value);
+            break;
+          case BIT:
+            writer.setBoolean((Boolean) value);
+            break;
+          default:
+            logger.warn("{} types are not implemented as implicit fields.", dataType);
+        }
+      }
+    }
+  }
+
+  public static class ImplicitColumn {
+    private final String fieldName;
+    private final MinorType dataType;
+    private final int columnIndex;
+    private final ScalarWriter writer;
+    private Object value;
+
+    public ImplicitColumn(String fieldName, MinorType dataType, RowSetLoader rowWriter) {
+      this.dataType = dataType;
+      this.fieldName = fieldName;
+      this.columnIndex = addImplicitColumnToSchema(this.fieldName, this.dataType, rowWriter);
+      this.writer = rowWriter.scalar(this.columnIndex);
+    }
+
+    public ImplicitColumn(String fieldName, MinorType dataType, RowSetLoader rowWriter, Object value) {
+      this.dataType = dataType;
+      this.fieldName = fieldName;
+      this.columnIndex = addImplicitColumnToSchema(this.fieldName, this.dataType, rowWriter);
+      this.writer = rowWriter.scalar(this.columnIndex);
+      this.value = value;
+    }
+
+    /**
+     * Adds an implicit column to the schema. Implicit columns are by default optional and excluded from wildcard
+     * queries.  This should be used for file metadata or other metadata that you want to be present in a query, but only if
+     * a user specifically asks for it.
+     *
+     * @param fieldName The name of the implicit column to be added.  Should start with an underscore
+     * @param type The minor type of the implicit field.  Currently only non-complex types are supported with this class
+     * @param rowWriter The RowSetLoader
+     * @return The index of the newly added column.
+     */
+    private int addImplicitColumnToSchema(String fieldName, MinorType type, RowSetLoader rowWriter) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(fieldName, type, DataMode.OPTIONAL);
+      colSchema.setBooleanProperty(ColumnMetadata.EXCLUDE_FROM_WILDCARD, true);
```

Comment:
I am interested in the usage of `setBooleanProperty(ColumnMetadata.EXCLUDE_FROM_WILDCARD, true)`.